### PR TITLE
fix(vercel): Catch 403s from Vercel 

### PR DESCRIPTION
**Pull request by mgaeta**
Fixes [SENTRY-NNT](https://sentry.io/organizations/sentry/issues/2238040441/), [API-1722](https://getsentry.atlassian.net/browse/API-1722).

The integration details GET endpoint is supposed to return data about integrations. When the integration is Vercel, we make some API calls to populate some of the data. Sometimes the API call fails. This PR catches those API failures and silently moves on with the other integration details.

### CODE REVIEW

File: src/sentry/integrations/vercel/integration.py [hunk 1]
```diff
@@ -142,18 +142,25 @@ def get_slug(self):
...
+            slug = ""
+
         base_url = "https://vercel.com/%s" % slug
-        vercel_projects = [
+
+        client = self.get_client()
+        return [
             {"value": p["id"], "label": p["name"], "url": "{}/{}".format(base_url, p["name"])}
-            for p in vercel_client.get_projects()
+            for p in client.get_projects()
```
Comment (scefali) [inline src/sentry/integrations/vercel/integration.py:157]: @mgaeta aren't these links going to be bogus if we use an empty string for the slug?

File: src/sentry/integrations/vercel/integration.py [hunk 2]
```diff
@@ -164,15 +171,16 @@ def get_organization_config(self):
...
             {
                 "name": "project_mappings",
                 "type": "project_mapper",
                 "mappedDropdown": {
-                    "items": vercel_projects,
+                    "items": self._get_vercel_projects(),
                     "placeholder": _("Vercel project..."),
                 },
-                "sentryProjects": sentry_projects,
+                "sentry_projects": self._get_sentry_projects(),
```
Comment (scefali) [inline src/sentry/integrations/vercel/integration.py:183]: @mgaeta why is this changing to snake case?

File: src/sentry/integrations/vercel/integration.py [hunk 3]
```diff
@@ -142,18 +142,25 @@ def get_slug(self):
             user = client.get_user()
             return user["username"]
 
-    def get_organization_config(self):
-        vercel_client = self.get_client()
-        # TODO: add try/catch if we get API failure
-        slug = self.get_slug()
+    def _get_vercel_projects(self):
+        """If the API returns a 403, just silently replace the slug."""
```
Comment (scefali) [inline src/sentry/integrations/vercel/integration.py:146]: @mgaeta if the API returns a 403, isn't everything going to be broken related to the integration?

File: src/sentry/integrations/vercel/integration.py (hunk 2 above)
Comment (mgaeta) [reply src/sentry/integrations/vercel/integration.py]: Good catch, reverted.

**Review by scefali — CHANGES_REQUESTED** [review]
When an integration is totally busted and we are loading up the configuration page for that specific integration, we cannot fail silently without telling the user. That's just going to lead to even more confusing result (ex: "I mapped my projects but they they longer appear, what's going on"?) A red error screen is better than hiding the fact it's broken since the user can still uninstall a busted integration.

**Comment by mgaeta** [comment]
I've updated `_get_vercel_projects` to silently return an empty list if there is an `APIError`. I can also update the `ProjectMapperField` to display an error in this case.

**Pull request closed without merging**